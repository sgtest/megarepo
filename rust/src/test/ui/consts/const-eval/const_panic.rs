#![feature(const_panic)]
#![allow(non_fmt_panic)]
#![crate_type = "lib"]

const MSG: &str = "hello";

const Z: () = std::panic!("cheese");
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out

const Z2: () = std::panic!();
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out

const Y: () = std::unreachable!();
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out

const X: () = std::unimplemented!();
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out
//
const W: () = std::panic!(MSG);
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out

const Z_CORE: () = core::panic!("cheese");
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out

const Z2_CORE: () = core::panic!();
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out

const Y_CORE: () = core::unreachable!();
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out

const X_CORE: () = core::unimplemented!();
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out

const W_CORE: () = core::panic!(MSG);
//~^ ERROR any use of this value will cause an error
//~| WARN this was previously accepted by the compiler but is being phased out
