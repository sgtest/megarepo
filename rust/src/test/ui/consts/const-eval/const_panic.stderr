error: any use of this value will cause an error
  --> $DIR/const_panic.rs:14:1
   |
LL | pub const Z: () = panic!("cheese");
   | ^^^^^^^^^^^^^^^^^^----------------^
   |                   |
   |                   the evaluated program panicked at 'cheese', $DIR/const_panic.rs:14:19
   |
   = note: #[deny(const_err)] on by default
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:17:1
   |
LL | pub const Y: () = unreachable!();
   | ^^^^^^^^^^^^^^^^^^--------------^
   |                   |
   |                   the evaluated program panicked at 'internal error: entered unreachable code', $DIR/const_panic.rs:17:19
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:20:1
   |
LL | pub const X: () = unimplemented!();
   | ^^^^^^^^^^^^^^^^^^----------------^
   |                   |
   |                   the evaluated program panicked at 'not yet implemented', $DIR/const_panic.rs:20:19
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to 3 previous errors

